samples/streamFileUpload.js

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on files with
+ * the Google Cloud Storage API.
+ *
+ * For more information, see the README.md under /storage and the documentation
+ * at https://cloud.google.com/storage/docs.
+ */
+
+function main(
+  bucketName = 'my-bucket',
+  filePath = './local/path/to/file.txt',
+  destFileName = 'file.txt'
+) {
+  // [START storage_stream_file_upload]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample
+   */
+  // The ID of your GCS bucket
+  // const bucketName = 'your-unique-bucket-name';
+
+  // The path to your file to upload
+  // const filePath = 'path/to/your/file';
+
+  // The new ID for your GCS file
+  // const destFileName = 'your-new-file-name';
+
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Import Node.js filesystem
+  const fs = require('fs');
+
+  // Creates a client
+  const storage = new Storage();
+
+  // Get a reference to the bucket
+  const myBucket = storage.bucket(bucketName);
+
+  // Create a reference to a file object
+  const file = myBucket.file(destFileName);
+
+  async function streamFileUpload() {
+    fs.createReadStream(filePath)
+      .pipe(file.createWriteStream())
+      .on('error', () => {
```

Comment:
I don't think we should catch errors like this in the sample. What do you think? In this case it will fail silently. Maybe just remove this clause altogether?